compiler/AST/ForLoop.cpp

```diff
@@ -77,6 +77,13 @@ static void tryToReplaceWithDirectRangeIterator(Expr* iteratorExpr)
   Expr* stride = NULL;
   if (CallExpr* call = toCallExpr(iteratorExpr))
   {
+    // if we have a 1D domain constructor, just grab the inner range and see if
+    // that can be optimized away
+    if (call->isNamed("chpl__buildDomainExpr") and call->numActuals() == 1)
```

Comment:
Elliot, I'm too lazy to check: Is there any chance we use this same call to create other singleton domains (like, I'm thinking, an associative domain with one element in it?)